File: client/src/app/models/balance.ts
import { Currency } from './currency';

export class Balance {
  constructor(public account: Account, public totals: Total[]) {}
}

type Total = {
  currency: Currency
  balance: number;
  toEurRate: number;
};

type Account = {
  reference: string;
  displayName: string;
};

File: server/src/main/java/com/invoicefair/server/domain/Balance.java
package com.invoicefair.server.domain;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.Arrays;
import java.util.List;

@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class Balance {
    @Getter private final Account account;
    @Getter private final List<Total> totals;

    public static Balance create(Account account, Total ...totals) {
        return new Balance(account, Arrays.asList(totals));
    }
}
